src/java/picard/analysis/artifacts/CollectSequencingArtifactMetrics.java

```diff
@@ -203,6 +212,14 @@ protected void acceptRead(final SAMRecord rec, final ReferenceSequence ref) {
                 final int readPos = block.getReadStart() + offset;
                 final int refPos = block.getReferenceStart() + offset;
 
+                // skip low BQ sites
+                final byte qual = readQuals[readPos - 1];
+                if (qual < MINIMUM_QUALITY_SCORE) continue;
+
+                // skip N bases in read
+                final char readBase = (char)readBases[readPos - 1];  // SAMRecord guarantees that all bases are already upper-cased
```

Comment:
Unless you're toUpper'ing the whole read earlier, `BAMRecord` guarantees all bases are upper case, but `SAMRecord` doesn't.